pkg/jaeger/query/query.go

```diff
@@ -41,45 +44,83 @@ func (p *Query) SpanWriter() spanstore.Writer {
 }
 
 func (p *Query) GetTrace(ctx context.Context, traceID model.TraceID) (*model.Trace, error) {
+	metrics.RequestsTotal.With(prometheus.Labels{"type": "trace", "handler": "get_trace", "code": ""}).Inc()
+	start := time.Now()
 	res, err := getTrace(ctx, p.conn, traceID)
 	if err == nil {
+		metrics.RequestsTotal.With(prometheus.Labels{"type": "trace", "handler": "get_trace", "code": "200"}).Inc()
```

Comment:
I think using `err == nil` is really uncommon and hurts readability since you usually handle `err != nil` cases. We should rewrite this and return early in the err check, would improve the code significantly.